### ISSUE TITLE
UnkownEndpoint error when trying to use plugin 

### ISSUE THREAD

**Issue by thescientist13** [MikaAK/s3-plugin-webpack#69]
- [ x] I have read and understood this plugin's README
- [ x] If filing a bug report, I have included my version of node and s3-plugin-webpack
- [ x] If filing a bug report, I have included which OS (including specific OS
  version) I am using.
- [ x] If filing a bug report, I have included a minimal test case that reproduces
  my issue.
- [ x] **I understand this is an open-source project staffed by someone with a job and
  that any help I receive is done on free time. I know I am not entitled to anything and will be polite and courteous.**
- [x ] **I understand my issue may be closed if it becomes obvious I didn't
  actually perform all of these steps or the issue is not with the library itself**

# Issue Details
I have been trying to use this plugin and although the S3 upload part does work (I can confirm files are uploaded), I get an  error and I don't see the Invalidation show up in my AWS Cloudfront Invalidation console for my Cloudfront Distribution. 

I made sure to check that I have an IAM role with the `CloudFrontFullAccess` policy.  

**tl;dr files are uploaded to S3 as expected, but there is an `UnknownEndpoint` error and no invalidation is recorded in my CloudFront dashboard.**

# Environment
```bash
$ node -v
v6.9.4
$ yarn -v
yarn install v0.21.3
[1/4] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 1.15s.
```

Using `"webpack-s3-plugin": "^1.0.0-rc.0"` on MacOS Sierra 10.12.3

# Steps to reproduce
Here is my webpack config
```javascript
// webpack.config.release.js
const prodConfig = require('./webpack.config.prod');
const S3Plugin = require('webpack-s3-plugin');
const webpackMerge = require('webpack-merge');

module.exports = webpackMerge(prodConfig, {

  plugins: [
    new S3Plugin({
      s3Options: {
        accessKeyId: 'xxx',
        secretAccessKey: 'xxx',
        region: 'us-east-1'
      },
      s3UploadOptions: {
        Bucket: 'xxx'
      },
      cdnizerOptions: {
        defaultCDNBase: 'xxx'
      },
      cloudfrontInvalidateOptions: {
        DistributionId: 'xxx',
        Items: ['/index.html']
      }
    })
  ]

});
```

I run `yarn run release`
```javascript
"release": "webpack --config ./webpack.config.release.js  --progress --profile --bail",
```

# Issue
And this is the error I'm getting.
```bash
ERROR in S3Plugin: UnknownEndpoint: Inaccessible host: `s3.amazonaws.com'. This service may not be available in the `us-east-1' region.
Child html-webpack-plugin for "index.html":
        + 1 hidden modules
Child favicons-webpack-plugin for "icons/stats.json":
               Asset     Size  Chunks             Chunk Names
    icons/stats.json  5.63 kB       0  [emitted]
        + 1 hidden modules
Uploading [>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 100% 0.0s
✨  Done in 605.86s.
```

Thanks for making this plugin and let me know if you need anymore information!

**Comment by aMoRRoMa** [comment]
I have same problem :(

**Comment by MikaAK** [comment]
Thanks for the feedback. I'll try to get to fixing this soon. For now, I would downgrade unless necessary. I'm not actually 100% sure how the cloudfront invalidation actually works so I will have to do some research and figure out how best to get around this. @flipxfx may be able to guess on this right away though

**Comment by flipxfx** [comment]
Weird that the error is shown before as the invalidation happens after right? This error seems to be related to s3. https://github.com/andrewrk/node-s3-client/issues/68#issuecomment-164413320

**Comment by thescientist13** [comment]
Doing some more testing locally and I think I have narrowed down the issue to my usage of `CopyPlugin` with a large directory from _node_modules_ for loading **CKEditor** as part of my Angular app's bundled.

```javascript
new CopyWebpackPlugin([{
  //TODO add using import?
  context: './node_modules/ckeditor/',
  from: '**/**',
  to: './assets/vendor/ckeditor/'
}]),
```

In reviewing options for optimizing the loading of **ckeditor** (which is about 15MB) I noticed a newer version **ng2-ckeditor** that was released that supports Angular's `NgModule`, which may provide a better bundling opportunity and reduce the dependency on `CopyPlugin` and greedy copying all files from that directory in _node_modules_

**Comment by thescientist13** [comment]
Yeah, refactoring my build seems to have resolved my issue.  Hope this can help someone else at least.
